packages/microsoft_kiota_http/lib/src/http_client_request_adapter.dart

```diff
@@ -254,29 +260,33 @@ class HttpClientRequestAdapter implements RequestAdapter {
       return null;
     }
 
-    switch (ModelType) {
-      case const (bool):
-        return rootNode.getBoolValue() as ModelType;
-      case const (int):
-        return rootNode.getIntValue() as ModelType;
-      case const (double):
-        return rootNode.getDoubleValue() as ModelType;
-      case const (String):
-        return rootNode.getStringValue() as ModelType;
-      case const (DateTime):
-        return rootNode.getDateTimeValue() as ModelType;
-      case const (DateOnly):
-        return rootNode.getDateOnlyValue() as ModelType;
-      case const (TimeOnly):
-        return rootNode.getTimeOnlyValue() as ModelType;
-      case const (Duration):
-        return rootNode.getDurationValue() as ModelType;
-      case const (UuidValue):
-        return rootNode.getGuidValue() as ModelType;
-      default:
-        throw ArgumentError(
-          'The type $ModelType is not supported for primitive deserialization',
-        );
+    if (_isExactly<ModelType, bool>() || _isExactly<ModelType, bool?>()) {
```

Comment:
Not anything critical, but considering that all uses of `_isExactly` use it for non-nullable and nullable, we could change it to

```dart
  @pragma('vm:prefer-inline')
  bool _isExactlyOrNullable<T, U>() => T == U || T == U?;
```

Reply:
Wait so Dart doesn't support nullable type literals but it does support nullable generic type literals?

---

Checked, doesn't work. Same issue as with other nullable literal types.

Reply:
Oh, that makes sense lol

Reply:
We could still abstract in a different way, if you feel it is worthy, but feel free to close it and do as you prefer.

Reply:
I mean, after inlining, it both results in the same code anyways 🙃

Reply:
> We could still abstract in a different way

You have another idea?

Reply:
> I mean, after inlining, it both results in the same code anyways 🙃

Yes, it's just that I feel like it's a little hard to read.

> You have another idea?

If you feel like it's worthy, we can workaround the workaround with:

```dart
Type typeOf<T>() => T;

bool _isExactly<T, U>() => T == U || T == typeOf<U?>();
```

But it's just a stylistic preference.

Reply:
Interestingly, there's a issue proposing to drop type literals completely in favor of a `typeOf` function.

https://github.com/dart-lang/language/issues/2393

Reply:
That's a good idea! The issue you linked also sounds interesting